File: java/src/main/java/webmis/modules/admin/UserPasswd.java
package webmis.modules.admin;

import java.util.HashMap;

import javax.servlet.http.HttpServletRequest;

import com.alibaba.fastjson.JSONObject;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import webmis.service.Base;
import webmis.service.AdminToken;
import webmis.library.Safety;
import webmis.model.User;
import webmis.util.Hash;

@RestController
@Controller("AdminUserPasswd")
@RequestMapping("/admin/user_passwd")
public class UserPasswd extends Base {

  /* 编辑 */
  @RequestMapping("edit")
  String Edit(@RequestBody JSONObject json, HttpServletRequest request) {
    HashMap<String,Object> res;
    // 参数
    String token = JsonName(json, "token");
    String passwd = JsonName(json, "passwd");
    String passwdNew = JsonName(json, "passwdNew");
    // 验证
    String msg = AdminToken.Verify(token, "");
    if(!msg.equals("")){
      res = new HashMap<String,Object>();
      res.put("code", 4001);
      res.put("msg", msg);
      return GetJSON(res);
    }
    if(passwd==passwdNew) {
      res = new HashMap<String,Object>();
      res.put("code", 4000);
      res.put("msg", "不能与原密码相同!");
      return GetJSON(res);
    }
    if(!Safety.IsRight("passwd", passwd) || !Safety.IsRight("passwd", passwdNew)) {
      res = new HashMap<String,Object>();
      res.put("code", 4000);
      res.put("msg", "密码为6～16位!");
      return GetJSON(res);
    }
    // 数据
    HashMap<String, Object> tData = AdminToken.Token(token);
    User model = new User();
    model.Columns("id");
    model.Where("id=? AND password=?", tData.get("uid").toString(), Hash.Md5(passwd));
    HashMap<String, Object> uData = model.FindFirst();
    if(uData.isEmpty()){
      res = new HashMap<String,Object>();
      res.put("code", 4000);
      res.put("msg", "当前密码错误!");
      return GetJSON(res);
    }
    HashMap<String, Object> upParam = new HashMap<String, Object>();
    upParam.put("password", Hash.Md5(passwdNew));
    model.Set(upParam);
    model.Where("id=?", tData.get("uid").toString());
    if(!model.Update()) {
      res = new HashMap<String,Object>();
      res.put("code", 5000);
      res.put("msg", "修改失败!");
      return GetJSON(res);
    }
    // 返回
    res = new HashMap<String,Object>();
    res.put("code", 0);
    res.put("msg", "成功");
    return GetJSON(res);
  }

}

File: gin/modules/admin/user_passwd.go
package admin

import (
	"webmis/library"
	"webmis/model"
	"webmis/service"
	"webmis/util"

	"github.com/gin-gonic/gin"
)

/* 用户密码 */
type UserPasswd struct {
	service.Base
}

/* 编辑 */
func (r UserPasswd) Edit(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token := r.JsonName(json, "token")
	passwd := r.JsonName(json, "passwd")
	passwdNew := r.JsonName(json, "passwdNew")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, "")
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if passwd == passwdNew {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "不能与原密码相同!"})
		return
	}
	if !(&library.Safety{}).IsRight("passwd", passwd) || !(&library.Safety{}).IsRight("passwd", passwdNew) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "密码为6～16位!"})
		return
	}
	// 数据
	tData := (&service.AdminToken{}).Token(token)
	model := (&model.User{}).New()
	model.Columns("id")
	model.Where("id=? AND password=?", tData["uid"], (&util.Hash{}).Md5(passwd))
	uData := model.FindFirst()
	if uData == nil {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "当前密码错误!"})
		return
	}
	model.Set(map[string]interface{}{"password": (&util.Hash{}).Md5(passwdNew)})
	model.Where("id=?", tData["uid"])
	if !model.Update() {
		r.GetJSON(c, gin.H{"code": 5000, "msg": "修改失败!"})
		return
	}
	// 返回
	r.GetJSON(c, gin.H{"code": 0, "msg": "成功"})
}
